AngularApp/src/app/shared/services/diagnostic-api.service.ts

```diff
@@ -36,7 +36,7 @@ export class DiagnosticApiService {
 
   public getDetectors(version: string, resourceId: string, body?: any): Observable<DetectorMetaData[]> {
     let path = `${version}${resourceId}/detectors`;
-    return this.invoke<DetectorResponse[]>(path, HttpMethod.POST, body).map(response => response.map(detector => detector.metadata));
+    return this.invoke<DetectorResponse[]>(path, HttpMethod.POST, body).retry(1).map(response => response.map(detector => detector.metadata));
```

Comment:
Not sure I understand the question, it will only retry if it fails. This call is pretty fast. This is just meant to solve the problem of the detector list not showing up.